app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala

```diff
@@ -47,76 +48,77 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
-            },
-            {
-              case (Eligible(_), Some(details)) ⇒
+            }, {
+              case UserSessionInfo.EligibleWithPayePersonalDetails(_, details) ⇒
                 Ok(views.html.you_are_eligible(details))
-              case (Ineligible(_), _) ⇒
+              case UserSessionInfo.Ineligible(_) ⇒
                 SeeOther(routes.StrideController.youAreNotEligible().url)
-              case (AlreadyHasAccount(_), _) ⇒
+              case UserSessionInfo.AlreadyHasAccount(_) ⇒
                 SeeOther(routes.StrideController.accountAlreadyExists().url)
-              case _ ⇒
-                logger.warn("unknown error during checking eligibility and pay-personal-details")
-                internalServerError()
             }
           )
         })
     )
   }(routes.StrideController.checkEligibilityAndGetPersonalInfo())
 
   def youAreNotEligible: Action[AnyContent] = authorisedFromStride { implicit request ⇒
-    checkSession(SeeOther(routes.StrideController.getEligibilityPage().url))
+    checkSession(_ ⇒ SeeOther(routes.StrideController.getEligibilityPage().url))
   }(routes.StrideController.youAreNotEligible())
 
   def youAreEligible: Action[AnyContent] = authorisedFromStride { implicit request ⇒
-    checkSession(SeeOther(routes.StrideController.getEligibilityPage().url))
+    checkSession(_ ⇒ SeeOther(routes.StrideController.getEligibilityPage().url))
   }(routes.StrideController.youAreEligible())
 
   def accountAlreadyExists: Action[AnyContent] = authorisedFromStride { implicit request ⇒
-    checkSession(SeeOther(routes.StrideController.getEligibilityPage().url))
+    checkSession(_ ⇒ SeeOther(routes.StrideController.getEligibilityPage().url))
   }(routes.StrideController.accountAlreadyExists())
 
+  def getTermsAndConditionsPage: Action[AnyContent] = authorisedFromStride { implicit request ⇒
+    checkSession(_ ⇒ SeeOther(routes.StrideController.getEligibilityPage().url),
+      checkIsEligible(_ ⇒ Ok(views.html.terms_and_conditions()))
+    )
+  }(routes.StrideController.getTermsAndConditionsPage())
+
+  private def checkIsEligible(ifEligible: EligibleWithPayePersonalDetails ⇒ Future[Result])(userInfo: UserSessionInfo): Future[Result] =
+    userInfo match {
+      case e: EligibleWithPayePersonalDetails ⇒
+        ifEligible(e)
+
+      case UserSessionInfo.Ineligible(_) ⇒
+        SeeOther(routes.StrideController.youAreNotEligible().url)
+
+      case UserSessionInfo.AlreadyHasAccount(_) ⇒
+        SeeOther(routes.StrideController.accountAlreadyExists().url)
+
+    }
+
   private def getPersonalDetails(r:           EligibilityCheckResult,
                                  ninoEncoded: String)(implicit hc: HeaderCarrier,
-                                                      request: Request[_]): EitherT[Future, String, Option[PayePersonalDetails]] = {
+                                                      request: Request[_]): EitherT[Future, String, UserSessionInfo] =
```

Comment:
minor: function name is  `getPersonalDetails ` but its returning  `UserSessionInfo`